### ISSUE TITLE
AWS API Gateway 'Method already exists' error when adding a new method to an existing endpoint

### ISSUE THREAD

**Issue by bradenhc** [serverless/serverless#7231]
# Bug Report

## Description
Deployment to AWS fails when adding a method on an existing endpoint created from a previous Serverless deployment (e.g. POST on `/events` exists and I want to add GET on the existing `/events` endpoint). Here is the output of `serverless --version`:

```
Framework Core: 1.61.2
Plugin: 3.2.7
SDK: 2.2.1
Components Core: 1.1.2
Components CLI: 1.4.0
```

I am running Node.js v12.13.1 on Ubuntu 18.04 LTS.

### What did you do?
I started by creating a simple Serverless application with a single HTTP POST event on the `/events` endpoint, as in the following YAML file:

```yaml
service:
  name: SimpleApplication
provider:
  name: aws
  stage: dev
  region: us-west-2
functions:
  PostFunction:
    handler: handler.post
    name: 'PostFunction-${self:provider.stage}'
    runtime: nodejs12.x
    events:
      - http:
          name: PostEvent
          path: /events
          method: post
```

The `handler.js` file looks like the following:

```js
'use strict';

module.exports = {
  post: require('./src/post')
};
```

And the `src/post.js` file looks like this:

```js
'use strict';

async function handler(event) {
  return {
    statusCode: 201,
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({ message: 'Success in Post!' })
  };
}

module.exports = handler;
```

This setup results in the following directory structure:

```
serverless.yml
handler.js
src/
 |_ post.js
```

From this state, running `serverless deploy` works as expected, successfully deploying the configuration. I then edit my `serverless.yml` file by adding a GET method to the `/events` endpoint:

```yaml
service:
  name: SimpleApplication
provider:
  name: aws
  stage: dev
  region: us-west-2
functions:
  PostFunction:
    handler: handler.post
    name: 'PostFunction-${self:provider.stage}'
    runtime: nodejs12.x
    events:
      - http:
          name: PostEvent
          path: /events
          method: post
  GetFunction:
    handler: handler.get
    name: 'GetFunction-${self:provider.stage}'
    runtime: nodejs12.x
    events:
      - http:
          name: GetEvent
          path: /events
          method: get

```

I also add the appropriate handlers:

**`handler.js`**
```js
'use strict';

module.exports = {
  post: require('./src/post'),
  get: require('./src/get')
};
```

**`src/get.js`**
```js
'use strict';

async function handler(event) {
  return {
    statusCode: 200,
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({ message: 'Success in Get!' })
  };
}

module.exports = handler;
```

All of this results in the following directory structure:

```
serverless.yml
handler.js
src/
 |_ post.js
 |_ get.js
```

### What happened?

With this new configuration, running `serverless deploy` fails with the following error from AWS:

```
ApiGatewayMethodGetEventPost - Method already exists for this resource
```

A couple of things I'e noticed:
1. The name of the method, `ApiGatewayMethodGetEventPost`, doesn't make sense because there is no POST method named `GetEvent`. `GetEvent` is GET. I'm assuming there is an error somehwere in the CloudFormation template generation.
1. It appears that Serverless is trying to recreate the POST method on the `/events` endpoint, which obviously doesn't work because it already exists.

### What should've happened?

The deployment should have succeeded, adding a new HTTP event endpoint for GET requests on the `/events` path.

### What's the content of your `serverless.yml` file?

For the successful deployment:

```yaml
service:
  name: SimpleApplication
provider:
  name: aws
  stage: dev
  region: us-west-2
functions:
  PostFunction:
    handler: handler.post
    name: 'PostFunction-${self:provider.stage}'
    runtime: nodejs12.x
    events:
      - http:
          name: PostEvent
          path: /events
          method: post
```

For the subsequent, failing deployment:

```yaml
service:
  name: SimpleApplication
provider:
  name: aws
  stage: dev
  region: us-west-2
functions:
  PostFunction:
    handler: handler.post
    name: 'PostFunction-${self:provider.stage}'
    runtime: nodejs12.x
    events:
      - http:
          name: PostEvent
          path: /events
          method: post
  GetFunction:
    handler: handler.get
    name: 'GetFunction-${self:provider.stage}'
    runtime: nodejs12.x
    events:
      - http:
          name: GetEvent
          path: /events
          method: get
```

### What's the output you get when you use the `SLS_DEBUG=*` environment variable (e.g. `SLS_DEBUG=* serverless deploy`)

This is the output from the failed attempt:

```text
Serverless: Load command interactiveCli
Serverless: Load command config
Serverless: Load command config:credentials
Serverless: Load command config:tabcompletion
Serverless: Load command config:tabcompletion:install
Serverless: Load command config:tabcompletion:uninstall
Serverless: Load command create
Serverless: Load command install
Serverless: Load command package
Serverless: Load command deploy
Serverless: Load command deploy:function
Serverless: Load command deploy:list
Serverless: Load command deploy:list:functions
Serverless: Load command invoke
Serverless: Load command invoke:local
Serverless: Load command info
Serverless: Load command logs
Serverless: Load command metrics
Serverless: Load command print
Serverless: Load command remove
Serverless: Load command rollback
Serverless: Load command rollback:function
Serverless: Load command slstats
Serverless: Load command plugin
Serverless: Load command plugin
Serverless: Load command plugin:install
Serverless: Load command plugin
Serverless: Load command plugin:uninstall
Serverless: Load command plugin
Serverless: Load command plugin:list
Serverless: Load command plugin
Serverless: Load command plugin:search
Serverless: Load command config
Serverless: Load command config:credentials
Serverless: Load command rollback
Serverless: Load command rollback:function
Serverless: Load command upgrade
Serverless: Load command uninstall
Serverless: Load command login
Serverless: Load command logout
Serverless: Load command generate-event
Serverless: Load command test
Serverless: Load command dashboard
Serverless: Invoke deploy
Serverless: Invoke package
Serverless: Invoke aws:common:validate
Serverless: Invoke aws:common:cleanupTempDir
Serverless: Packaging service...
Serverless: Excluding development dependencies...
Serverless: Invoke aws:package:finalize
Serverless: Invoke aws:common:moveArtifactsToPackage
Serverless: Invoke aws:common:validate
Serverless: Invoke aws:deploy:deploy
Serverless: [AWS cloudformation 200 0.196s 0 retries] describeStacks({ StackName: 'SimpleApplication-dev' })
Serverless: [AWS cloudformation 200 0.19s 0 retries] describeStackResource({
  StackName: 'SimpleApplication-dev',
  LogicalResourceId: 'ServerlessDeploymentBucket'
})
Serverless: [AWS s3 200 0.222s 0 retries] listObjectsV2({
  Bucket: 'simpleapplication-dev-serverlessdeploymentbucket-1f417i66uulh2',
  Prefix: 'serverless/SimpleApplication/dev'
})
Serverless: [AWS s3 200 0.183s 0 retries] headObject({
  Bucket: 'simpleapplication-dev-serverlessdeploymentbucket-1f417i66uulh2',
  Key: 'serverless/SimpleApplication/dev/1579375103080-2020-01-18T19:18:23.080Z/SimpleApplication.zip'
})
Serverless: [AWS s3 200 0.222s 0 retries] headObject({
  Bucket: 'simpleapplication-dev-serverlessdeploymentbucket-1f417i66uulh2',
  Key: 'serverless/SimpleApplication/dev/1579375103080-2020-01-18T19:18:23.080Z/compiled-cloudformation-template.json'
})
Serverless: [AWS lambda 200 0.24s 0 retries] getFunction({ FunctionName: 'PostFunction-dev' })
Serverless: [AWS lambda 404 0.238s 0 retries] getFunction({ FunctionName: 'GetFunction-dev' })
Serverless: [AWS sts 200 0.391s 0 retries] getCallerIdentity({})
Serverless: Uploading CloudFormation file to S3...
Serverless: [AWS s3 200 0.197s 0 retries] putObject({
  Body: <Buffer 7b 22 41 57 53 54 65 6d 70 6c 61 74 65 46 6f 72 6d 61 74 56 65 72 73 69 6f 6e 22 3a 22 32 30 31 30 2d 30 39 2d 30 39 22 2c 22 44 65 73 63 72 69 70 74 ... 6529 more bytes>,
  Bucket: 'simpleapplication-dev-serverlessdeploymentbucket-1f417i66uulh2',
  Key: 'serverless/SimpleApplication/dev/1579375209414-2020-01-18T19:20:09.414Z/compiled-cloudformation-template.json',
  ContentType: 'application/json',
  Metadata: { filesha256: 'ZnuacYDaqF+nmUnhpJFCMjiBU/Xia8RhCs6vUd9aacs=' }
})
Serverless: Uploading artifacts...
Serverless: Uploading service SimpleApplication.zip file to S3 (5.55 KB)...
Serverless: [AWS s3 200 0.209s 0 retries] putObject({
  Body: <Buffer 50 4b 03 04 14 00 08 00 08 00 00 00 21 00 00 00 00 00 00 00 00 00 00 00 00 00 15 00 00 00 2e 76 73 63 6f 64 65 2f 73 65 74 74 69 6e 67 73 2e 6a 73 6f ... 5630 more bytes>,
  Bucket: 'simpleapplication-dev-serverlessdeploymentbucket-1f417i66uulh2',
  Key: 'serverless/SimpleApplication/dev/1579375209414-2020-01-18T19:20:09.414Z/SimpleApplication.zip',
  ContentType: 'application/zip',
  Metadata: { filesha256: 'miBYkmCgWh6jz0gOwLGm5y8gSwfWr/T9c6PI+1WrZ0E=' }
})
Serverless: Validating template...
Serverless: [AWS cloudformation 200 1.005s 0 retries] validateTemplate({
  TemplateURL: 'https://s3.amazonaws.com/simpleapplication-dev-serverlessdeploymentbucket-1f417i66uulh2/serverless/SimpleApplication/dev/1579375209414-2020-01-18T19:20:09.414Z/compiled-cloudformation-template.json'
})
Serverless: Updating Stack...
Serverless: [AWS cloudformation 200 0.602s 0 retries] updateStack({
  StackName: 'SimpleApplication-dev',
  Capabilities: [ 'CAPABILITY_IAM', 'CAPABILITY_NAMED_IAM', [length]: 2 ],
  Parameters: [ [length]: 0 ],
  TemplateURL: 'https://s3.amazonaws.com/simpleapplication-dev-serverlessdeploymentbucket-1f417i66uulh2/serverless/SimpleApplication/dev/1579375209414-2020-01-18T19:20:09.414Z/compiled-cloudformation-template.json',
  Tags: [ { Key: 'STAGE', Value: 'dev' }, [length]: 1 ]
})
Serverless: Checking Stack update progress...
Serverless: [AWS cloudformation 200 0.448s 0 retries] describeStackEvents({
  StackName: 'arn:aws:cloudformation:us-west-2:796471679983:stack/SimpleApplication-dev/4bb462f0-3a27-11ea-af6c-0696a78c5fe8'
})
.....Serverless: [AWS cloudformation 200 0.872s 0 retries] describeStackEvents({
  StackName: 'arn:aws:cloudformation:us-west-2:796471679983:stack/SimpleApplication-dev/4bb462f0-3a27-11ea-af6c-0696a78c5fe8'
})
Serverless: [AWS cloudformation 200 0.384s 0 retries] describeStackEvents({
  StackName: 'arn:aws:cloudformation:us-west-2:796471679983:stack/SimpleApplication-dev/4bb462f0-3a27-11ea-af6c-0696a78c5fe8'
})
Serverless: [AWS cloudformation 200 0.335s 0 retries] describeStackEvents({
  StackName: 'arn:aws:cloudformation:us-west-2:796471679983:stack/SimpleApplication-dev/4bb462f0-3a27-11ea-af6c-0696a78c5fe8'
})
.......Serverless: [AWS cloudformation 200 0.377s 0 retries] describeStackEvents({
  StackName: 'arn:aws:cloudformation:us-west-2:796471679983:stack/SimpleApplication-dev/4bb462f0-3a27-11ea-af6c-0696a78c5fe8'
})
............
Serverless: Operation failed!
Serverless: View the full error output: https://us-west-2.console.aws.amazon.com/cloudformation/home?region=us-west-2#/stack/detail?stackId=arn%3Aaws%3Acloudformation%3Aus-west-2%3A796471679983%3Astack%2FSimpleApplication-dev%2F4bb462f0-3a27-11ea-af6c-0696a78c5fe8
 
  Serverless Error ---------------------------------------
 
  ServerlessError: An error occurred: ApiGatewayMethodGetEventPost - Method already exists for this resource (Service: AmazonApiGateway; Status Code: 409; Error Code: ConflictException; Request ID: 34e9b045-ef19-4555-9df5-4bf4c3804e7b).
      at /home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/lib/plugins/aws/lib/monitorStack.js:125:33
  From previous event:
      at AwsDeploy.monitorStack (/home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/lib/plugins/aws/lib/monitorStack.js:28:12)
      at /home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/lib/plugins/aws/lib/updateStack.js:107:28
  From previous event:
      at AwsDeploy.update (/home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/lib/plugins/aws/lib/updateStack.js:107:8)
  From previous event:
      at AwsDeploy.<anonymous> (/home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/lib/plugins/aws/lib/updateStack.js:121:35)
  From previous event:
      at AwsDeploy.updateStack (/home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/lib/plugins/aws/lib/updateStack.js:117:33)
  From previous event:
      at AwsDeploy.<anonymous> (/home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/lib/plugins/aws/deploy/index.js:127:39)
  From previous event:
      at Object.aws:deploy:deploy:updateStack [as hook] (/home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/lib/plugins/aws/deploy/index.js:123:30)
      at /home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/lib/classes/PluginManager.js:490:55
  From previous event:
      at PluginManager.invoke (/home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/lib/classes/PluginManager.js:490:22)
      at PluginManager.spawn (/home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/lib/classes/PluginManager.js:510:17)
      at AwsDeploy.<anonymous> (/home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/lib/plugins/aws/deploy/index.js:93:48)
  From previous event:
      at Object.deploy:deploy [as hook] (/home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/lib/plugins/aws/deploy/index.js:89:30)
      at /home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/lib/classes/PluginManager.js:490:55
  From previous event:
      at PluginManager.invoke (/home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/lib/classes/PluginManager.js:490:22)
      at /home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/lib/classes/PluginManager.js:525:24
  From previous event:
      at PluginManager.run (/home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/lib/classes/PluginManager.js:525:8)
      at /home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/lib/Serverless.js:133:33
      at processImmediate (internal/timers.js:439:21)
      at process.topLevelDomainCallback (domain.js:130:23)
  From previous event:
      at Serverless.run (/home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/lib/Serverless.js:120:74)
      at /home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/bin/serverless.js:75:30
      at /home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/node_modules/graceful-fs/graceful-fs.js:136:16
      at /home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/node_modules/graceful-fs/graceful-fs.js:57:14
      at FSReqCallback.oncomplete (fs.js:146:23)
  From previous event:
      at /home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/bin/serverless.js:75:8
      at processImmediate (internal/timers.js:439:21)
      at process.topLevelDomainCallback (domain.js:130:23)
  From previous event:
      at Object.<anonymous> (/home/braden/.nvm/versions/node/v12.13.1/lib/node_modules/serverless/bin/serverless.js:64:4)
      at Module._compile (internal/modules/cjs/loader.js:959:30)
      at Object.Module._extensions..js (internal/modules/cjs/loader.js:995:10)
      at Module.load (internal/modules/cjs/loader.js:815:32)
      at Function.Module._load (internal/modules/cjs/loader.js:727:14)
      at Function.Module.runMain (internal/modules/cjs/loader.js:1047:10)
      at internal/main/run_main_module.js:17:11
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com
 
  Your Environment Information ---------------------------
     Operating System:          linux
     Node Version:              12.13.1
     Framework Version:         1.61.2
     Plugin Version:            3.2.7
     SDK Version:               2.2.1
     Components Core Version:   1.1.2
     Components CLI Version:    1.4.0
```

### Similar or dependent issues

None that I can find

**Comment by FanFataL** [comment]
@bradenhc Hi,

TLDR: Tou have used unsuported `name` parameter in the `http` event. Just remove it.

The bug appears because you have used unsuported `name` parameter in the `http` event...
```
    events:
      - http:
          name: PostEvent
          path: /events
          method: post
```

In your case sls made 1 resource:
 - `ApiGatewayMethodPostEventPost`
Then you added second one and sls made 2 resources:
 - `ApiGatewayMethodGetEventPost` (sls just takes first resource with the same path)
 - `ApiGatewayMethodGetEventGet`
Adn because cloudformation first create resource then cleanup rest the error occures.

If you do configuration without `name` param all would work in normal way:
Sls would create one resource with autogenerated name:
 - `ApiGatewayMethodEventsPost`
Then after you added second one and sls would made 2 resources:
 - `ApiGatewayMethodEventsPost`
 - `ApiGatewayMethodEventsGet`

As you can see there is no replacing resources in cloudformation in the second case.

**Comment by medikoo** [comment]
As @FanFataL pointed, `name` is not a supported configuration property.

There's a side effect of using it, that it overrides resource name in a template, and that's clearly a framework bug (technically using `name` should be ineffective, and possibly throw a warning after #6562 is taken in)

**Comment by bradenhc** [comment]
Awesome, thank you @FanFataL and @medikoo for the quick response. I removed the `name` property and the deployment worked as expected.

Looking forward to #6562 being taken in. For now I'm pretty booked, but I may be able to help if I end up with some spare time down the road and there still isn't a PR for it.

**Comment by medikoo** [comment]
I'm closing this, as currently configuration is validated, and warning is shown when the unsupported property is found
